### PR TITLE
ci: Add github workflow POLICY

### PR DESCRIPTION
**Pull request by phlax**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: .github/workflows/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+- `workflow_run`
+- `workflow_dispatch`
+
+Do not use these trigger events unless they are required.
+
+## Restrict access to trusted workflows
+
+It is important to restrict who can trigger these types of workflow.
+
+Do not allow any bots or app users to do so, unless this is specifically required.
```
Comment (phlax) [inline .github/workflows/README.md:25]: i need to clarify how to do this

**Comment by phlax** [comment]
related to #26170

File: .github/workflows/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+In particular this can effect:
+
+- `pull_request_target`
+- `workflow_run`
+- `workflow_dispatch`
+
+Do not use these trigger events unless they are required.
+
+## Restrict access to trusted workflows
```
Comment (phlax) [inline .github/workflows/README.md:21]: i think this only affects `workflow_dispatch`

File: .github/workflows/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+## Trusted/untrusted CI jobs
+
+If a trusted workflow is used to run untrusted code, then the entire job that runs this code
+should be treated as untrusted.
+
+In this case, it is **essential** to ensure:
+
+- restrictive global read permissions
+- no global write permissions
```
Comment (phlax) [inline .github/workflows/README.md:35]: probably these should be required everywhere

Comment (joycebrum) [reply .github/workflows/README.md:35]: Agreed!

File: .github/workflows/README.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+- no write permissions in the untrusted job
+- no secrets in the untrusted job
```
Comment (joycebrum) [inline .github/workflows/README.md:37]:
```suggestion
- no global secrets
- no secrets in the untrusted job
```

File: .github/workflows/README.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+Do not allow any bots or app users to do so, unless this is specifically required.
+
+## Trusted/untrusted CI jobs
+
+If a trusted workflow is used to run untrusted code, then the entire job that runs this code
+should be treated as untrusted.
+
+In this case, it is **essential** to ensure:
+
+- restrictive global read permissions
```
Comment (joycebrum) [inline .github/workflows/README.md:34]: I think it allso should be required everywhere.

**Comment by phlax** [comment]
i have renamed README -> POLICY to give it a bit more authority, and updated the guidance

**Comment by phlax** [comment]
@wbpcode lets land this - im sure it will iterate, but it will give us a decent basis to review gh wfs i think

**Review by wbpcode — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**